src/webgl/material.js

```diff
@@ -29,6 +29,7 @@ require('./p5.Texture');
  * vertex and fragment shader files.
  */
 p5.prototype.loadShader = function(vertFilename, fragFilename) {
+  this._assert3d('loadShader');
```

Comment:
Looking through the manual test examples, I got this when looking at the toon shader example...

```
Uncaught TypeError: Cannot read property 'isP3D' of undefined
    at p5._assert3d (p5.js:72953)
    at p5.loadShader (p5.js:69876)
    at p5.<anonymous> (p5.js:46111)
    at preload (sketch.js:4)
    at p5.<anonymous> (p5.js:46076)
    at new p5 (p5.js:46363)
    at _globalInit (p5.js:48341)
```

During preload, this._renderer isn't yet defined on the p5 instance. If it's not obvious how to fix this, my suggestion is to remove this line from the commit so that custom shaders continue to work with preload and we can merge this patch promptly. Then we can open a new issue to resolve this for loadShader as a unique case.

Reply:
you're right. this assertion is completely invalid. i don't think it's every necessary here.